R/get_draws.R

```diff
@@ -228,43 +239,58 @@ get_draws.data.frame <- function(x,
         by = "date"
       ) |>
       dplyr::ungroup() |>
-      dplyr::rename("observed_value" = "count") |>
       dplyr::select(
         "date",
-        "draw",
-        "observed_value",
         "pred_value",
+        "draw",
         "total_pop"
       )
   } else {
     NULL
   }
 
   subpop_rt_draws <- if (what_ok["subpop_rt"]) {
+    # Site index to corresponding site and subpopulation size
+    site_spine <- x |>
+      dplyr::distinct(.data$site, .data$site_index, .data$site_pop) |>
+      dplyr::mutate(subpop = glue::glue("Site: {as.factor(.data$site)}")) |>
+      dplyr::rename(
+        "subpop_pop" = "site_pop",
+        "subpop_index" = "site_index"
+      )
+
+    # If we added an auxiliary site, we need to adjust indexing accordingly
+    if (length(stan_data_list$subpop_size) > nrow(site_spine)) {
+      subpop_spine <- site_spine |>
+        dplyr::mutate(subpop_index = .data$subpop_index + 1) |>
+        dplyr::bind_rows(tibble::tibble(
+          site = NA,
+          subpop_index = 1,
+          subpop_pop = stan_data_list$subpop_size[1],
+          subpop = "remainder of pop"
+        ))
+    } else {
+      subpop_spine <- site_spine
+    }
```

Comment:
Now that we're making a strong distinction between subpopulations and sites, I think it's clearer and safer to do this with an explicit `subpop_index` and `site_index` defined in pre-processing.  This also makes it possible to adjust whether auxiliary site == reference site if we wish.
 
Related: should we not just calculate the spines in pre-processing and have them as attributes of the `wwinferencefit` object?

Reply:
Just to make sure I am understanding correctly, in this implementation you are suggesting that we have an explicit subpop_index which here would be 1:(site_index) + 1 if theres an aux site and equal to site index if not. 

We moved away from passing the spines as attributes of `wwinferencefit`, with the idea being that now that we are returning the input data, any of the spines can be calculated from the data (plus passing several spines seemed just a bit clunky...)